packages/eslint-plugin/src/rules/unbound-method.ts

```diff
@@ -0,0 +1,123 @@
+import { TSESTree, AST_NODE_TYPES } from '@typescript-eslint/typescript-estree';
+import * as tsutils from 'tsutils';
+import * as ts from 'typescript';
+
+import * as util from '../util';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+interface Config {
+  ignoreStatic: boolean;
+}
+
+type Options = [Config];
+
+type MessageIds = 'unbound';
+
+export default util.createRule<Options, MessageIds>({
+  name: 'unbound-method',
+  meta: {
+    docs: {
+      category: 'Best Practices',
+      description:
+        'Enforces unbound methods are called with their expected scope.',
+      tslintName: 'no-unbound-method',
+      recommended: 'error'
+    },
+    messages: {
+      unbound:
+        'Avoid referencing unbound methods which may cause unintentional scoping of `this`.'
+    },
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          ignoreStatic: {
+            type: 'boolean'
+          }
+        },
+        additionalProperties: false
+      }
+    ],
+    type: 'problem'
+  },
+  defaultOptions: [
+    {
+      ignoreStatic: false
+    }
+  ],
+  create(context, [{ ignoreStatic }]) {
+    const parserServices = util.getParserServices(context);
+    const checker = parserServices.program.getTypeChecker();
+
+    return {
+      MemberExpression(node) {
```

Comment:
If we decide to use `AST_NODE_TYPES` at all, this node selector should be `AST_NODE_TYPES.MemberExpression` with a computed property.